kedro-org/kedro-viz
# [KED-2962] Export run data as CSV

## Description

resolves [KED-2962](https://jira.quantumblack.com/browse/KED-2962).

This PR is to build out the 'export run' feature, which consolidates the selected run metadata and tracking data as shown on the experiment detail panel into a CSV file for download. 

<img width="1631" alt="image" src="https://user-images.githubusercontent.com/27775658/155435243-d043ac31-d33f-4e70-b62a-e1daf86114d9.png">


## Development notes

I have made the following changes:

-  moved the fetching of the run metadata and the tracking data up in the experiment-tracking wrapper level to avoid redundancy in network requests. 
- set up a set of util functions to format the run metadata and run tracking data into an array for download
- set up the 'export csv' button to expose this function

Please note that, in case you are opening the csv file in excel, the alignment of the value would vary - please note that this is simply a csv export and is not optimized for excel. 

## QA notes

Please try downloading the csv export and check the download file. The name should be name as `rundata-(runID1)-(runID2).csv`, and the csv should contain all data that is displayed in the details view. 

## Checklist



<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/757"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
